x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -132,20 +177,5 @@ export class RuleDataClient implements IRuleDataClient {
         }
       }
     }
-
-    const { body: simulateResponse } = await clusterClient.transport.request({
-      method: 'POST',
-      path: `/_index_template/_simulate_index/${concreteIndexName}`,
-    });
-
-    const mappings: estypes.MappingTypeMapping = simulateResponse.template.mappings;
-
-    if (isEmpty(mappings)) {
-      throw new Error(
-        'No mappings would be generated for this index, possibly due to failed/misconfigured bootstrapping'
-      );
-    }
```

Comment:
I'm not sure if this should be removed. I don't see the same check in `rolloverAliasIfNeeded `, and I think we should be explicit about this, because teams new to RAC will run into it - I did, too, and it is a little annoying to figure out and repair.

Reply:
This sort of check seems like it belongs with the template bootstrapping logic IMO. At this point we've already created the index, so simulating it seems redundant. Can we move this check so it happens sometime before the index is created, maybe at index template creation time?

Reply:
I'm fine with moving it, but I do would like to keep it, _somewhere_. But let's figure out first where this all needs to happen.

Reply:
There should be always some common mappings specified (currently they are specified in templates installed in `init()` - `TECHNICAL_COMPONENT_TEMPLATE_NAME` and `ECS_COMPONENT_TEMPLATE_NAME`). We just have to make sure that final index templates always reference (`composed_of`) these common component templates.

I'd say this is a matter of having an API that doesn't provide 100% flexibility in index bootstrapping and makes sure that index template is always created and it always references the common templates - doesn't matter how the client of the library uses it.

Reply:
Moved this check to index template creation time in https://github.com/elastic/kibana/pull/102586/commits/4ccf63a2ca45878e56f7de569ad65cdd427ab91f